### PR TITLE
feat: support get output from cmd offline query

### LINKED ISSUE #1480: Support query offline data in cli and show result
Now we can only get job info when executing offline sql in cmd.

### PR DESCRIPTION
**Pull request by tobegit3hub**
* Resolve https://github.com/4paradigm/OpenMLDB/issues/1480 .

Here is the example when executing offline sql in command-line.

Screen Shot 2022-03-18 at 7 18 25 PM

### CODE REVIEW

**Comment by tobegit3hub** [comment]
Usage: 

```
SET SESSION sync_job = true;
SET SESSION execute_mode = "offline";

SELECT 1;
```

image

File: java/openmldb-taskmanager/src/main/scala/com/_4paradigm/openmldb/taskmanager/OpenmldbBatchjobManager.scala [hunk 1]
```diff
@@ -39,15 +39,17 @@ object OpenmldbBatchjobManager {
    * Run the SparkSQL job and save output to specified path.
    *
    * @param sql the SQL text
-   * @param outputPath the output path
    * @return the Yarn AppId in String format
    */
-  def runBatchSql(sql: String, outputPath: String, sparkConf: java.util.Map[String, String], defaultDb: String): JobInfo = {
+  def runBatchSql(sql: String, sparkConf: java.util.Map[String, String], defaultDb: String): String = {
+    // tobe
```
Comment (tobegit3hub) [inline java/openmldb-taskmanager/src/main/scala/com/_4paradigm/openmldb/taskmanager/OpenmldbBatchjobManager.scala]: Remove debug line.

File: src/sdk/sql_cluster_router.cc [hunk 2]
```diff
@@ -2293,20 +2304,38 @@ std::shared_ptr<hybridse::sdk::ResultSet> SQLClusterRouter::ExecuteSQL(const std
...
+
+                if (IsSyncJob()) {
+                    // Run offline sql and wait to get output
+                    std::string output;
+                    auto base_status = ExecuteOfflineQueryGetOutput(sql, config, db, output);
+                    if (base_status.OK()) {
+                        *status = {};
+                        // Print the output from job output
+                        // TODO: return result set if want to format the output
+                        std::cout << output << std::endl;
```
Comment (tobegit3hub) [inline src/sdk/sql_cluster_router.cc]: Maybe set to `ResultSetSQL::MakeResultSet({FORMAT_STRING_KEY}, {value}, status);`.

File: src/sdk/sql_cluster_router.cc [hunk 3]
```diff
@@ -2293,20 +2304,38 @@ std::shared_ptr<hybridse::sdk::ResultSet> SQLClusterRouter::ExecuteSQL(const std
...
-                    if (job_info.id() > 0) {
-                        std::stringstream ss;
-                        ::openmldb::cmd::PrintJobInfos({job_info}, ss);
-                        std::vector<std::string> value = {ss.str()};
-                        return ResultSetSQL::MakeResultSet({FORMAT_STRING_KEY}, {value}, status);
+
+                if (IsSyncJob()) {
+                    // Run offline sql and wait to get output
+                    std::string output;
+                    auto base_status = ExecuteOfflineQueryGetOutput(sql, config, db, output);
```
Comment (dl239) [inline src/sdk/sql_cluster_router.cc:2312]: if only used here, rm declaration from sql_router.h

**Review by dl239 — APPROVED** [review]

**Review by vagetablechicken — APPROVED** [review]

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**